### ISSUE TITLE
Move "force the use of hwloc" commits to v2.x

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#869]
So far, these commits are involved:

* open-mpi/ompi@d97bc29102112198270452fc37b7dfedcef9ce75 
* open-mpi/ompi@f782a7640e4ea36ae632a82238c8b2402565428e 
* open-mpi/ompi@794ee4a604dd114dd1332ac26ce7bc9d6e8b537e

As of this writing (8am US Eastern Sep 5 2015), there's several PMIX-related conflicts when trying to merge the first of these commits to v2.x.  I'm assuming the latest PMIX stuff hasn't been PR'ed over to v2.x -- when they are, these commits should apply cleanly.

**Comment by hppritcha** [comment]
@jsquyres do we care about the conflicts related to the lack of the treematch PRs in v2.x?

**Comment by hppritcha** [comment]
We probably need commit 7cb6c2fc too.

**Comment by jsquyres** [comment]
7cb6c2f probably got swept up in open-mpi/ompi-release#671.

I'll file a PR for this one to start moving it forward.

**Comment by jsquyres** [comment]
@rhc54 I actually ran into a bunch of ORTE conflicts when I tried to merge the first commit into v2.x that are not immediately obvious to me how to fix... :-(

Can you have a look / take a shot at creating a PR?

```
        both modified:   opal/mca/btl/openib/btl_openib_component.c
        both modified:   orte/mca/ess/base/ess_base_std_orted.c
        both modified:   orte/mca/ess/hnp/ess_hnp_module.c
        both modified:   orte/mca/ess/pmi/ess_pmi_module.c
        both modified:   orte/mca/ras/simulator/ras_sim_module.c
        both modified:   orte/runtime/data_type_support/orte_dt_print_fns.c
```

**Comment by rhc54** [comment]
Okay - I'll give it a shot

**Comment by hppritcha** [comment]
It looks like we're missing  commits, at least for orte/mca/ras/simulator/ras_sim_module.c:
ed93154e43ed7c7fdf09205c42922e1790a8a67
023936e84bce80ab90ae1af5438163432f894d3f

Also there were white-space changes that were not exactly equivalent.

**Comment by hppritcha** [comment]
I opened a PR open-mpi/ompi-release#729 to merge 023936e from 7/29 (or thereabouts) into ompi-release. 

I would prefer to see 023936e and ed93154 go in as separate PRs to ompi-release.

**Comment by hppritcha** [comment]
Well 729 is botched.  Need commit ed93154 in too.

**Comment by jsquyres** [comment]
This one might become significantly easier once open-mpi/ompi-release#730 is applied (to the v2.x branch).

**Comment by jsquyres** [comment]
This was fixed by open-mpi/ompi-release#766
